src/main/java/uk/gov/ida/matchingserviceadapter/MatchingServiceAdapterModule.java

```diff
@@ -184,9 +185,12 @@ public EidasMatchingDatasetUnmarshaller getEidasMatchingDatasetUnmarshaller() {
 
     @Provides
     @Singleton
-    public EidasUnsignedMatchingDatasetUnmarshaller getEidasUnsignedMatchingDatasetUnmarshaller(SecretKeyDecryptorFactory secretKeyDecryptorFactory) {
+    public EidasUnsignedMatchingDatasetUnmarshaller getEidasUnsignedMatchingDatasetUnmarshaller(
+            SecretKeyDecryptorFactory secretKeyDecryptorFactory,
+            MetadataResolverRepository eidasMetadataResolverRepository) {
         StringToOpenSamlObjectTransformer<Response> stringtoOpenSamlObjectTransformer = new CoreTransformersFactory().getStringtoOpenSamlObjectTransformer(new ResponseSizeValidator());
-        return new EidasUnsignedMatchingDatasetUnmarshaller(secretKeyDecryptorFactory, stringtoOpenSamlObjectTransformer);
+        EidasValidatorFactory eidasValidatorFactory = new EidasValidatorFactory(eidasMetadataResolverRepository);
+        return new EidasUnsignedMatchingDatasetUnmarshaller(secretKeyDecryptorFactory, stringtoOpenSamlObjectTransformer, eidasValidatorFactory);
```

Comment:
It feels like there should be more to do here. Does `EidasUnsignedMatchingDatasetUnmarshaller` already have a constructor that accepts an `EidasValidatorFactory`?

Reply:
It does since alphagov/verify-saml-libs#96 PR was merged 😄